conans/test/functional/toolchain/test_adjust_compiler_linux.py

```diff
@@ -0,0 +1,289 @@
+        client.run("build ..")
+        testcase.assertIn("Using Conan toolchain", client.out)
+
+    cmake_cache = load(os.path.join(build_directory, "CMakeCache.txt"))
+    return client.out, cmake_cache, build_directory, None
+
+
+def _compile_cache_workflow(testcase, client, profile, use_toolchain):
+    # Compile the app in the cache
+    pref = client.create(ref=ConanFileReference.loads("app/version@user/channel"), conanfile=None,
+                         args=" --profile={} -o use_toolchain={}".format(profile, use_toolchain))
+    if use_toolchain:
+        testcase.assertIn("Using Conan toolchain", client.out)
+
+    package_layout = client.cache.package_layout(pref.ref)
+    build_directory = package_layout.build(pref)
+    cmake_cache = load(os.path.join(build_directory, "CMakeCache.txt"))
+    return client.out, cmake_cache, build_directory, package_layout.package(pref)
+
+
+def compile_cache_workflow_with_toolchain(testcase, client, profile):
+    return _compile_cache_workflow(testcase, client, profile, use_toolchain=True)
+
+
+def compile_cache_workflow_without_toolchain(testcase, client, profile):
+    return _compile_cache_workflow(testcase, client, profile, use_toolchain=False)
+
+
+def compile_cmake_workflow(testcase, client, profile):
+    build_directory = os.path.join(client.current_folder, "build")
+    rmdir(build_directory)
+    with client.chdir(build_directory):
+        client.run("install .. --profile={}".format(profile))
+        client.run_command("cmake .. -DCMAKE_TOOLCHAIN_FILE={}".format(CMakeToolchain.filename))
+        testcase.assertIn("Using Conan toolchain", client.out)
+
+    cmake_cache = load(os.path.join(build_directory, "CMakeCache.txt"))
+    return client.out, cmake_cache, build_directory, None
+
+
+@parameterized_class([{"function": compile_cache_workflow_without_toolchain, "use_toolchain": False,
+                       "in_cache": True},
+                      {"function": compile_cache_workflow_with_toolchain, "use_toolchain": True,
+                       "in_cache": True},
+                      {"function": compile_local_workflow, "use_toolchain": True, "in_cache": False},
+                      {"function": compile_cmake_workflow, "use_toolchain": True, "in_cache": False},
+                      ])
+@attr("toolchain")
+@unittest.skipUnless(platform.system() == "Linux", "Only linux")
+class AdjustAutoTestCase(unittest.TestCase):
+    """
+        Check that it works adjusting values from the toolchain file
+    """
+
+    _conanfile = textwrap.dedent("""
+        from conans import ConanFile, CMake, CMakeToolchain
+
+        class App(ConanFile):
+            name = "app"
+            version = "version"
+            settings = "os", "arch", "compiler", "build_type"
+            exports = "*.cpp", "*.txt"
+            generators = {% if use_toolchain %}"cmake_find_package"{% else %}"cmake"{% endif %}
+            options = {"use_toolchain": [True, False], "fPIC": [True, False]}
```

Comment:
This reads very weird, with the use_toolchain as an option, it shouldn't be.

I guess the problem is that this test is complicated, I am not sure why it is testing it without the toolchain too.

Reply:
Until all the changes are done I need/want to test without the toolchain too to **be sure that both workflows end up with the same variables and same values**. Once there are no more changes to do, we can remove some tests and refactor its code a little bit.